### PR TITLE
patch release 0.4 as 0.4.1

### PR DESCRIPTION
**Pull request by cnpetra**
This PR addresses a couple of small items forgotten when releasing 0.4. Addresses all the issues reported by issue #207 and comments made in PR#206.

Plan to do a new release as 0.4.1 (as opposed to updating 0.4, which I am not sure it's possible).

### CODE REVIEW

**Comment by ashermancinelli** [comment]
We can update the tag if you want, but it seems better to just have a new patch release like you suggest.￼

**Review by ashermancinelli — APPROVED** [review]

File: doc/src_usermanual/nlpsparse.tex [hunk 1]
```diff
@@ -104,7 +104,6 @@ \subsubsection{Specifying the optimization problem}
 \begin{lstlisting}
 bool eval_Jac_cons(const long long& n, const long long& m, 
 			          const double* x, bool new_x,
-			          const long long& nsparse, const long long& ndense,
```
Comment (pelesh) [inline doc/src_usermanual/nlpsparse.tex:107]: Why was this line removed? Was this a copy-paste error?

**Comment by pelesh** [comment]
In `techrep_main.tex`, line 266, C++ requirement should be C++11.

**Comment by pelesh** [comment]
On page 22, memory space option, the "default" option is sequential code. It is the no-RAJA option. The other three match Umpire memory spaces HOST, DEVICE and UM.

**Review by pelesh — CHANGES_REQUESTED** [review]
I would suggest fixing C++ standard and memory space options description in the manual. Other than that, it is ready to merge.

File: doc/src_usermanual/nlpsparse.tex (hunk 1 above)
Comment (cnpetra) [reply doc/src_usermanual/nlpsparse.tex:107]: was copy-paste error indeed

**Review by nychiang — APPROVED** [review]
